### PR TITLE
feat: add actions for installing and restoring node_modules

### PR DESCRIPTION
**Pull request by chrstph-dvx**
Add actions for `node_modules` caching (install and restore)

### CODE REVIEW

File: node-module-restore-cache/action.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  using: "composite"
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Restore "node_modules" from cache
+      uses: martijnhols/actions-cache/restore@v3
+      with:
+        path: "**/node_modules"
+        key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
```
Comment (spsjvc) [inline node-module-restore-cache/action.yml:13]: Ideally we could specify the install command as an input, and have this be the default. 
The reason is easier switching to npm/pnpm if we ever need to.

File: node-module-save-cache/action.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Check cache for "node_modules"
+      id: cache
+      uses: martijnhols/actions-cache/check@v3
+      with:
+        path: "**/node_modules"
+        key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
```
Comment (spsjvc) [inline node-module-save-cache/action.yml:14]: Same comment around specifying as input.

**Review by spsjvc — COMMENTED** [review]
Let's also reorganize these actions like this:

```
node-modules/
--install/
----action.yml
--restore/
----action.yml
```

Makes it cleaner to use:
```
OffchainLabs/actions/node-modules/install@main
OffchainLabs/actions/node-modules/restore@main
```

File: node-module-save-cache/action.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    - name: Install dependencies
+      shell: bash
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: yarn install --frozen-lockfile
```
[the remark this answers is not in the document]
Comment (chrstph-dvx) [reply node-module-save-cache/action.yml]: Moved to input with default

File: node-module-restore-cache/action.yml (hunk 1 above)
Comment (chrstph-dvx) [reply node-module-restore-cache/action.yml]: Moved to input with default

File: node-module-restore-cache/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+Restore node_modules from cache
+
+## Usage
+
+In workflows:
+
+```yml
+jobs:
+  install:
+    name: Install node modules
```
[the remark this answers is not in the document]
Comment (chrstph-dvx) [reply node-module-restore-cache/README.md]: Renamed

File: node-modules/install/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+In workflows:
+
+```yml
+jobs:
+  install:
+    name: Install node modules
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install node_modules
+        uses: OffchainLabs/actions/node-module-save-cache@main
```
Comment (spsjvc) [inline node-modules/install/README.md:16]: This is outdated

File: node-modules/restore/README.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+```yml
+jobs:
+  install:
+    name: Restore node modules
+    needs: [install] # Name of the jobs installing node_modules through node-module-save-cache action
+    runs-on: ubuntu-latest
+    steps:
+      - name: Restore cache
+        uses: OffchainLabs/actions/node-module-restore-cache@main
```
Comment (spsjvc) [inline node-modules/restore/README.md:17]: This is outdated

File: node-modules/restore/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+Restore node_modules from cache
+
+## Usage
+
+In workflows:
+
+```yml
+jobs:
+  install:
```
Comment (spsjvc) [inline node-modules/restore/README.md:11]: We should rename this to anything else, maybe `example` to reduce confusion

**Review by spsjvc — APPROVED** [review]

**Pull request merged**